packages/api/src/controllers/animalController.js

```diff
@@ -65,12 +74,32 @@ const animalController = {
               .send('Should only send either default_breed_id or custom_breed_id');
           }
 
+          if (animal.default_type_id) {
+            const defaultType = await DefaultAnimalTypeModel.query().findById(
+              animal.default_type_id,
+            );
+
+            if (!defaultType) {
+              trx.rollback();
+              return res.status(400).send('default_type_id has invalid value');
+            }
+          }
+
+          if (animal.custom_type_id) {
+            const customType = await CustomAnimalTypeModel.query().findById(animal.default_type_id);
```

Comment:
This should be `.findById(animal.custom_type_id);`, right? I wonder why this doesn't affect any tests...

At a quick glance, the changes look good overall! I will spend some more time later on!